lib/parser/lexer.rl

```diff
@@ -1026,8 +1029,15 @@ class Parser::Lexer
           @herebody_s = current_literal.saved_herebody_s
         end
 
+        @cond.lexpop
+        if @version < 24
+          @cmdarg.lexpop
+        else
+          @cmdarg.pop
+        end
+
         fhold;
-        fnext *stack_pop;
```

Comment:
Is the stack still used anywhere? If not then `stack_pop` should be removed.

Reply:
Yes:
```
  expr_variable := |*
      global_var
      => {
        if    tok =~ /^\$([1-9][0-9]*)$/
          emit(:tNTH_REF, tok(@ts + 1).to_i)
        elsif tok =~ /^\$([&`'+])$/
          emit(:tBACK_REF)
        else
          emit(:tGVAR)
        end

        fnext *stack_pop; fbreak;
      };

      class_var_v
      => {
        if tok =~ /^@@[0-9]/
          diagnostic :error, :cvar_name, { :name => tok }
        end

        emit(:tCVAR)
        fnext *stack_pop; fbreak;
      };

      instance_var_v
      => {
        if tok =~ /^@[0-9]/
          diagnostic :error, :ivar_name, { :name => tok }
        end

        emit(:tIVAR)
        fnext *stack_pop; fbreak;
      };
  *|;
```

because there are still a couple of `fcall`-sin the code. From what I see none of them emit tokens that change `cmdarg` in the grammar.